### ISSUE TITLE
amazon makeBucket endpoint support not correct anymore

### ISSUE THREAD

**Issue by pszabop** [minio/minio-js#619]
I'm using Amazon S3 with the region us-east-2 and attempting to call makeBucket.

manually specifying 'us-east-2' to both the client initialization and makeBucket results in S3 error: `The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'us-west-1`. packet capture shows the library attempting to talk to `s3-us-west-1-w.amazonaws.com`.

I think maybe the problem is that the error name returned for the '[retry to the correct region](https://github.com/minio/minio-js/blob/fcc055944bcc2065b2678e3061b34d065ce2528c/src/main/minio.js#L520)' is not what is expected.  You are expecting `e.name == AuthorizationHeaderMalformed`.  Down lower the stack is converting this (AFAICT) to `S3Error` (that's the name in the exception).

Stack trace and error message follows:
```
"message":"The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'us-west-1'","name":"S3Error","stack":"S3Error: The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'us-west-1'\n    at Object.parseError (/usr/src/app/node_modules/minio/dist/main/xml-parsers.js:58:11)\n    at /usr/src/app/node_modules/minio/dist/main/transformers.js:118:22\n 
```

version is `3.1.3`

**Comment by krishnasrinivas** [comment]
@pszabop the following code worked fine:
```
var Minio = require('./minio-js')

// find out your s3 end point here:
// http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region
var s3Client = new Minio.Client({
  endPoint: "s3.amazonaws.com",
  accessKey: 'XXXXXXXXXXXXXXXXX',
  secretKey: 'XXXXXXXXXXXXXXXXXXXXXXX',
  region: 'us-east-2'
})

s3Client.traceOn()
s3Client.makeBucket('krisrand123', 'us-east-2', function(e) {
  if (e) {
    return console.log(e)
  }
  console.log("Success")
})

```
What minio-js version are you using?

**Comment by deekoder** [comment]
@pszabop  have you been able to makeBucket. We think this is working fine. Please confirm if its still not working for you.

**Comment by pszabop** [comment]
@deekoder I got it to work by commenting out the region in `Minio.Client` and setting the region in `makeBucket`.

> Minio-js abstracts region settings from the developers.

I suggest you address this in the documentation, because it's not clear when I need to specify the region and when I don't need to.   The auto-region discovery AFAICT isn't even discussed in the documentation.  I had to read the code to figure out what you were up to.   (basically newbie mistake that could be prevented)

For example, `makeBucket` requires a region but `Minio.Client` does not.   But AFAICT bucket namespaces in most cases are a global namespace across regions...
